lib/rules/property-no-deprecated/index.mjs

```diff
@@ -0,0 +1,126 @@
+import { isRegExp, isString } from '../../utils/validateTypes.mjs';
+import isStandardSyntaxDeclaration from '../../utils/isStandardSyntaxDeclaration.mjs';
+import optionsMatches from '../../utils/optionsMatches.mjs';
+import report from '../../utils/report.mjs';
+import ruleMessages from '../../utils/ruleMessages.mjs';
+import validateOptions from '../../utils/validateOptions.mjs';
+
+const ruleName = 'property-no-deprecated';
+
+const messages = ruleMessages(ruleName, {
+	expected: (unfixed, fixed) => `Expected "${unfixed}" to be "${fixed}"`,
+	rejected: (property) => `Unexpected deprecated property "${property}"`,
+});
+
+const meta = {
+	url: 'https://stylelint.io/user-guide/rules/property-no-deprecated',
+	fixable: true,
+};
+
+/** @type {Record<string, string | null>} */
+const DEPRECATED_PROPS_REMAP = {
+	'-khtml-box-flex': 'flex-grow',
+	'-khtml-box-ordinal-group': 'order',
+	'-khtml-box-sizing': 'box-sizing',
+	'-moz-box-flex': 'flex-grow',
+	'-moz-box-ordinal-group': 'order',
+	'-moz-box-sizing': 'box-sizing',
+	'-moz-column-gap': 'column-gap',
+	'-moz-user-select': 'user-select',
+	'-ms-user-select': 'user-select',
+	'-webkit-box-align': 'align-items',
+	'-webkit-box-flex': 'flex-grow',
+	'-webkit-box-ordinal-group': 'order',
+	'-webkit-box-orient': null,
+	'-webkit-box-pack': null,
+	'-webkit-box-sizing': 'box-sizing',
+	'-webkit-column-gap': 'column-gap',
+	'-webkit-user-select': 'user-select',
```

Comment:
```suggestion
	'-webkit-box-align': 'align-items',
	'-webkit-box-flex': 'flex-grow',
	'-webkit-box-ordinal-group': 'order',
	'-webkit-box-orient': null,
	'-webkit-box-pack': null,
	'-webkit-box-sizing': 'box-sizing',
	'-webkit-column-gap': 'column-gap',
```

The role of this rule is not related to the usage of prefixes.
i.e. prefixed property !== deprecated property
Hence I have remove the `user-select` mappings.